runtime/permissions/mod.rs

```diff
@@ -3107,54 +2647,35 @@ mod tests {
   fn test_revoke() {
```

Comment:
Should we now expect the right side of some of these assertions to be `PermissionState::GrantedPartial`? I expected it.

Also, if my understanding of `UnaryPermission.prompt` is correct, there shouldn't be any `PermissionState::Prompt` here. Is this right?

Reply:
> Should we now expect the right side of some of these assertions to be `PermissionState::GrantedPartial`? I expected it.

Nope, revoke behaviour is upwards recursing as it was before and is unaffected when there are no `--deny` flags. See https://deno.land/manual@v1.35.0/runtime/permission_apis#revoke-permissions. There are two reasons for this:
- There was no such thing as `GrantedPartial` until this PR. `read: /foo` always implied `read: /foo/bar`.
- The use case people brought up were to `--allow-read=./`, do some things with `./<files>`, revoke access to `./` while preserving access to `./public/` and then dynamic import dependencies.

Though, the other interpretation of `revoke()` can be argued for with similar use cases. i.e. `--allow-read=./`, do some things with `./private/`, revoke access downwards to `./private/` while preserving access to `./<files>`. Maybe that could be a different API like `Deno.permissions.deny()`.